# Sleep Study Dataset Analysis

The goal for this analysis was to understand human sleep behaviour using the "sleepstudy.csv" dataset. The dataset has information about the sleep duration, efficieny, REM, deep, and light sleep percentages of the participants. As well as some lifestyle factors like caffeine and alcohol consumption, exercise frequency, and smoking status. Using this data, we have the goal to understand how demographic facotrs and habits in the daily life influence sleep quality and duration. 

This analysis will include statistics, visualizations, and correlations to provide insights into which factors are most strongly associated with better or worse sleep. Understanding the patterns in the dataset can help with knowing healthier habits for sleep and overall lifestyle.

In [3]:
#Load Libraries
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [4]:
#Load csv file and check if it loaded correctly
df = pd.read_csv("sleepstudy.csv")
df.head()

ID  Age  Gender           Bedtime WakeupTime  SleepDuration  \
0   1   80  Female   2025-09-30 7:32    49:16.7       6.283241   
1   2   24    Male  2025-06-29 20:59    09:10.2       7.155613   
2   3   37    Male  2025-12-24 21:28    31:34.3       6.050627   
3   4   68  Female   2025-02-22 0:25    26:37.0       7.017791   
4   5   58    Male  2025-09-02 12:31    17:46.3       8.764793   

   SleepEfficiency  REMSleepPercentage  DeepSleepPercentage  \
0             0.57                  15                   35   
1             0.91                  29                   68   
2             0.58                  15                   35   
3             0.88                  28                   44   
4             0.95                  28                   40   

   LightSleepPercentage  Awakenings  CaffeineConsumption  AlcoholConsumption  \
0                    50           0                 25.0                   1   
1                     3           4                 50.0                   0   
2                    50           3                 50.0                   0   
3                    28           1                 50.0                   0   
4                    32           4                 25.0                   4   

  SmokingStatus  ExerciseFrequency  
0           Yes                  1  
1            No                  2  
2            No                  5  
3           Yes                  4  
4            No                  4

In [5]:
#Summary Statistics
df.describe(include = "all")

ID          Age Gender          Bedtime WakeupTime  \
count   1000.000000  1000.000000   1000             1000       1000   
unique          NaN          NaN      2             1000        990   
top             NaN          NaN   Male  2025-09-30 7:32    42:25.2   
freq            NaN          NaN    513                1          2   
mean     500.500000    50.125000    NaN              NaN        NaN   
std      288.819436    18.076397    NaN              NaN        NaN   
min        1.000000    18.000000    NaN              NaN        NaN   
25%      250.750000    34.000000    NaN              NaN        NaN   
50%      500.500000    51.000000    NaN              NaN        NaN   
75%      750.250000    65.000000    NaN              NaN        NaN   
max     1000.000000    80.000000    NaN              NaN        NaN   

        SleepDuration  SleepEfficiency  REMSleepPercentage  \
count     1000.000000      1000.000000         1000.000000   
unique            NaN              NaN                 NaN   
top               NaN              NaN                 NaN   
freq              NaN              NaN                 NaN   
mean         7.027443         0.720150           22.460000   
std          1.135365         0.128139            4.589059   
min          5.002335         0.500000           15.000000   
25%          6.076980         0.610000           18.000000   
50%          6.981505         0.720000           22.000000   
75%          8.039587         0.830000           27.000000   
max          8.998650         0.950000           30.000000   

        DeepSleepPercentage  LightSleepPercentage   Awakenings  \
count           1000.000000           1000.000000  1000.000000   
unique                  NaN                   NaN          NaN   
top                     NaN                   NaN          NaN   
freq                    NaN                   NaN          NaN   
mean              47.357000             30.183000     2.539000   
std               13.097293             13.788968     1.726691   
min               25.000000              1.000000     0.000000   
25%               36.000000             18.000000     1.000000   
50%               47.000000             30.000000     3.000000   
75%               59.000000             42.000000     4.000000   
max               70.000000             59.000000     5.000000   

        CaffeineConsumption  AlcoholConsumption SmokingStatus  \
count            744.000000         1000.000000          1000   
unique                  NaN                 NaN             2   
top                     NaN                 NaN           Yes   
freq                    NaN                 NaN           513   
mean              25.168011            2.560000           NaN   
std               20.446036            1.746267           NaN   
min                0.000000            0.000000           NaN   
25%                0.000000            1.000000           NaN   
50%               25.000000            3.000000           NaN   
75%               50.000000            4.000000           NaN   
max               50.000000            5.000000           NaN   

        ExerciseFrequency  
count         1000.000000  
unique                NaN  
top                   NaN  
freq                  NaN  
mean             2.565000  
std              1.712515  
min              0.000000  
25%              1.000000  
50%              3.000000  
75%              4.000000  
max              5.000000

In [6]:
#Count missing values in each column
df.isnull().sum()

ID                        0
Age                       0
Gender                    0
Bedtime                   0
WakeupTime                0
SleepDuration             0
SleepEfficiency           0
REMSleepPercentage        0
DeepSleepPercentage       0
LightSleepPercentage      0
Awakenings                0
CaffeineConsumption     256
AlcoholConsumption        0
SmokingStatus             0
ExerciseFrequency         0
dtype: int64

In [7]:
#Shape of dataset to help with Basic Summary
df.shape

(1000, 15)

In [15]:
#Figuring out the percentage of smokers for descriptive insights

smoker_percentage = (df['SmokingStatus'] == 'Yes').mean() * 100
print(round(smoker_percentage, 2))


51.3


## Basic Summary

This dataset has info on participants sleep paterns and lifestyle habits. Columns include SleepDuration, SleepEfficiency, REMSleepPercentage, DeepSleepPercentage, LightSleepPercentage, lifestyle habits like CaffeineConsumption, AlcoholConsumption, SmokingStatus, and ExerciseFrequency, along with Age and Gender. There are 1000 columns and 15 rows with 256 entries in "CaffeineConsumption" being blank.

## Descriptive Statistics

### Age

Participants range from 18-80 years old, wiht a median age of 51. The IQR shows that most participants are on the older side (34-65)

### Sleep Duration

On average participants sleep about 7 hours per night (mean = 7.027 hours), with a range of 5 to 9 hours across the whole dataset. Most of the dataset falls between the 6-8 hour range (the IQR).

### Sleep Efficiency

Average sleep efficency is around 72% with most participants between 61% and 83% (the 25th and 75th percentile respect). Some participants however, have very low sleep efficency (50%), while some reach a very high efficiency (95%).

### Sleep Stages

REM Sleep (~22% of total sleep):

REM (Rapid Eye Movement) sleep is the stage of sleep where dreaming occurs, characterized by high brain activity, increased heart rate, and temporary paralysis of major muscles. REM sleep is CRUCIAL for high quality sleep with the optimal amount comprising of 20-25% of your total sleep duration (essentially too much is bad). Participants spend roughs a fifth of their sleep in REM, which is normal.

Source: National Institutes of Health (https://www.nhlbi.nih.gov/health/sleep/stages-of-sleep and https://pmc.ncbi.nlm.nih.gov/articles/PMC2847051/)



Deep Sleep (~47% of total sleep):

Deep sleep is the most restorative stage of sleep. It helps the body recover, strengthens the immunge system, and supports muscle and tissue repair. Deep sleep is the largest portion of sleep. This number indicates participants are getting enough restorative sleep even when the total sleep duration varies.

Source: Sleep Foundation (https://www.sleepfoundation.org/stages-of-sleep/deep-sleep)

Light Sleep (~30% of total sleep):

Light sleep is the stage between waking up and deep sleep. Easier to wake up from and plays a role in physical recover and memory consolidation but is lest restorative than deep sleep. Participants spend about 30% which is normal.

Source: CLMSleep (https://www.clmsleep.com/light-sleep/)

### Awakenings

Participants experience 0-5 awakenings per night, with a median of 3, suggesting sleep interruptions are common. Upon research, sleepless nights are  common with approximately 2 in 5 Canadians experiencing sleepless nights according to Stats Canada.

Source: Statistics Canada (https://www.statcan.gc.ca/o1/en/plus/1653-cant-sleep-count-sheep)

### Lifestyle Factors

Exercise Frequency: Most participants exercise 0-5 days per week (median 3). More frequent exercise is generally linked to better sleep quality

Caffeine Consumption: Caffeine intake varies widely (0-50 mg/day). Higher caffeince use may reduce sleep duration or efficiency

Alcohol Consumption: Alcohol consumption ranges 0-5 ounces per day. Higher alcohol can interrupt sleep cycles and ruin sleep quality

Smoking Status: About half of the participants being smokers. Nicotine can negatively effect sleep duration and efficiency